### PR TITLE
adding some helper methods

### PR DESCRIPTION
**Pull request by acharb**
[fetching fee stats](https://stellarorg.atlassian.net/browse/WAL-468?atlOrigin=eyJpIjoiZDY5N2JkYjZlYTZkNDI3MGE4MTQ2MGFjZWRiOTdkNmMiLCJwIjoiaiJ9)

[resolve fed address](https://stellarorg.atlassian.net/browse/WAL-505?atlOrigin=eyJpIjoiYjQ4ODJhNjFlMjAzNGI2MDk3NTljODg0OTk2NGE5ODAiLCJwIjoiaiJ9)

[add friendbot helper](https://stellarorg.atlassian.net/browse/WAL-1216?atlOrigin=eyJpIjoiZGY3M2IxY2ZkZjVjNDVmZWJjNzE0YWIzM2MyNzcxNTkiLCJwIjoiaiJ9)

### CODE REVIEW

File: src/walletSdk/Horizon/Stellar.ts [hunk 1]
```diff
@@ -224,4 +226,35 @@ export class Stellar {
...
   }
+
+  /**
+   * Returns the recommended fee (stroops) to use in a transaction based on the current
+   * stellar network fee stats.
+   * @returns {string} The recommended fee amount in stroops.
+   */
+  async getRecommendedFee(): Promise<string> {
+    const stats = await this.server.feeStats();
+    return stats.max_fee.mode;
```
Comment (acharb) [inline src/walletSdk/Horizon/Stellar.ts:237]: this is how we're recommending fees in freighter, and I think makes sense

File: test/transaction.test.ts [hunk 2]
```diff
@@ -313,7 +313,9 @@ describe("Path Payment", () => {
     const txBuilder = await stellar.transaction({
       sourceAddress: sourceKp,
     });
-    const txn = txBuilder.swap(new NativeAssetId(), usdcAsset, ".1").build();
+    const txn = txBuilder
+      .swap(new NativeAssetId(), new NativeAssetId(), ".1")
```
Comment (acharb) [inline test/transaction.test.ts:317]: unrelated. Test is failing bc of testnet reset and no offers avail yet. Just using native -> native here instead, we already have path pay tests between 2 different assets, which swap is using under the hood

File: src/walletSdk/Horizon/Stellar.ts [hunk 3]
```diff
@@ -224,4 +226,35 @@ export class Stellar {
...
+  }
+
+  /**
+   * Resolves a federation address into a stellar address.
+   * @see {@link https://developers.stellar.org/docs/encyclopedia/federation}
+   * @param {string} fedAddress - The federation address (eg. jed*stellar.org).
+   * @returns {string} The stellar address associated with the federation address.
+   */
+  async resolveFederation(fedAddress: string): Promise<string> {
```
[the remark this answers is not in the document]
Comment (acharb) [reply src/walletSdk/Horizon/Stellar.ts]: ya that's fair, it is just a wrapper sgtm to remove
https://github.com/stellar/typescript-wallet-sdk/pull/82/commits/12351b35ea8731efa2faafafd4a6262db24621a3

File: src/walletSdk/Horizon/Stellar.ts [hunk 4]
```diff
@@ -224,4 +226,35 @@ export class Stellar {
...
+    return resp.account_id;
+  }
+
+  /**
+   * Funds an account on the stellar test network. If it is already funded then call will error.
+   * Please note: only funds on the testnet network.
+   * @see {@link https://developers.stellar.org/docs/fundamentals-and-concepts/testnet-and-pubnet#friendbot}
+   * @param {string} address - The stellar address.
+   */
+  async fundAccount(address: string) {
```
[the remark this answers is not in the document]
Comment (acharb) [reply src/walletSdk/Horizon/Stellar.ts]: 👍 https://github.com/stellar/typescript-wallet-sdk/pull/82/commits/46dfef43ed7838e9e33a5466ab1a00820aeeadef

**Review by Ifropc — APPROVED** [review]

**Pull request merged**